src/main/java/cz/inventi/kpj/spring/springintroduction/persistence/GreetingConfiguration.java

```diff
@@ -0,0 +1,22 @@
+package cz.inventi.kpj.spring.springintroduction.persistence;
+
+import cz.inventi.kpj.spring.springintroduction.persistence.GreetingConfigurationProperties;
+import cz.inventi.kpj.spring.springintroduction.persistence.GreetingConstantRepositoryImpl;
+import cz.inventi.kpj.spring.springintroduction.persistence.GreetingRepository;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties(GreetingConfigurationProperties.class)
+@ComponentScan(basePackages = "cz.inventi.kpj.spring.springintroduction")
```

Comment:
You are using component scan two times in your application with same base packages... It will probably work but it is not a good practice. It doesn't make sense in general ... 

I am marking this practice as completed.